plugins/aws_ecr/lib/samson_aws_ecr/samson_plugin.rb

```diff
@@ -0,0 +1,32 @@
+require 'aws-sdk-core'
+
+module SamsonAwsEcr
+  class Engine < Rails::Engine
+    AMAZON_REGISTRY = /\A.*\.dkr.ecr.(?<region>[\w\-]+).amazonaws.com\z/
+    mattr_accessor :ecr_client
+    mattr_accessor(:credentials_expire_at) { Time.current }
+
+    initializer :ecr_client do
+      if match = AMAZON_REGISTRY.match(ENV['DOCKER_REGISTRY'])
+        SamsonAwsEcr::Engine.ecr_client = Aws::ECR::Client.new(region: match['region'])
+      end
+    end
+
+  end
+end
+
+Samson::Hooks.callback :before_docker_build do
+  begin
+    if SamsonAwsEcr::Engine.ecr_client && SamsonAwsEcr::Engine.credentials_expire_at < Time.current
```

Comment:
I'm not sure how the AWS credential work. But, if they haven't expired, AWS returns the same credentials (that will have the same expires_at)... so... I'm not sure if that would be of any help. I could wait for the credentials to expire (but it seems too hackish)